frontend/test/resolution/testPoi.cpp

```diff
@@ -749,6 +749,121 @@ static void test6() {
   assert(rACallGeneric == rBCallGeneric);
 }
 
+// check that parenless non-method calls do not use POI
+static void test7() {
+  printf("test7\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  auto path = UniqueString::get(context, "input.chpl");
+  std::string contents = R""""(
+    module Library {
+      proc callFoo(type t) {
+        foo;
+      }
+    }
+
+    module Application {
+      use Library;
+
+      proc foo { return 1; }
+
+      proc main() {
+        callFoo(int);
+      }
+    }
+   )"""";
+
+  setFileText(context, path, contents);
+
+  const ModuleVec& vec = parseToplevel(context, path);
+  assert(vec.size() == 2);
+  auto Lib = vec[0]->toModule();
+  auto App = vec[1]->toModule();
+  assert(Lib);
+  assert(Lib->numStmts() == 1);
+  assert(App);
+  assert(App->numStmts() == 3);
+
+  auto main = App->stmt(2)->toFunction();
+  assert(main);
+  auto callCallFoo = main->stmt(0)->toCall();
+  assert(callCallFoo);
+  auto callFoo = Lib->stmt(0)->toFunction();
+  assert(callFoo);
+  auto foo = callFoo->stmt(0)->toIdentifier();
+  assert(foo);
+
+  auto rMain = resolveConcreteFunction(context, main->id());
+  assert(rMain);
+
+  auto rCallFoo = resolveOnlyCandidate(context, rMain->byAst(callCallFoo));
+  assert(rCallFoo);
```

Comment:
It's probably best to have someone more familiar with the dyno test set up take a look to ensure this test is correct, but to someone unfamiliar with the code - it seems like we should assert that we know `App->stmt(1)->toFunction()` is also true, right?  Otherwise we could get false positives for it not being found later.  After all, you verify the parenless method version is a function in test 8

Reply:
I added such a check.